30413/Stoica Tudor Mihai/Assignement 1/main.c

```diff
@@ -0,0 +1,343 @@
+        free(element);
+        element = first;
+    }
+    last=NULL;
+}
+
+void deleteFirst()
+{
+    nrElements--;
+    if(first==NULL || last==NULL)
+    {
+        return;
+    }
+
+    first=first->next;
+}
+
+void deleteLast()
+{
+    nrElements--;
+    if(first==NULL || last==NULL)
+    {
+        // printf("List is emtpy\n");
+        return;
+    }
+
+    if(first->next==NULL)
+    {
+        first=last=NULL;
+        return;
+    }
+
+    node * element=first;
+
+    while(element->next!=last)
+    {
+        element=element->next;
+    }
+
+    last=element;
+    last->next=NULL;
+    //printf("%d",last->info);
+}
+
+void PRINT_ALL (FILE *o)
+{
+    if(first==NULL || last==NULL)
+    {
+        fprintf(o,"List is emtpy\n");
+        return;
+    }
+
+    node * element=first;
+
+    while(element!=NULL)
+    {
+        fprintf(o,"%d ",element->info);
+        element=element->next;
+    }
+    fprintf(o,"\n");
+}
+
+void deleteElement(int data)
+{
+    nrElements--;
+    while(first->info==data)
+    {
+        first=first->next;
+    }
+
+    node * element=first->next;
+    node * previous=first;
+
+    while(element!=NULL)
+    {
+        if(element->info==data)
+        {
+            previous->next=element->next;
+        }
+
+        else
+        {
+            previous=element;
+        }
+        element=element->next;
+    }
+}
+
+void PRINT_F(int nr,FILE *o)
+{
+    node * element=first;
+
+    for(int i=0; i<nr && element!=NULL; i++)
+    {
+        fprintf(o,"%d ",element->info);
+        element=element->next;
+    }
+
+    fprintf(o,"\n");
+}
+
+void PRINT_L(int nr,FILE *o)
+{
+    if(nr>nrElements)
+    {
+        PRINT_ALL(o);
+        return;
+    }
+
+    node * element=first;
+
+    for(int i=0; i<nrElements-nr+1 && element!=NULL; i++)
+    {
+        element=element->next;
+    }
+
+    while(element!=NULL)
+    {
+        fprintf(o,"%d ",element->info);
+        element=element->next;
+    }
+    fprintf(o,"\n");
+
+}
+
+void PRINT_ALLL ()
+{
+    if(first==NULL || last==NULL)
+    {
+        printf("List is emtpy\n");
+        return;
+    }
+
+    node * element=first;
+
+    while(element!=NULL)
+    {
+        printf("%d ",element->info);
+        element=element->next;
+    }
+    printf("\n");
+}
+
+
+void readInstructions(FILE *f,FILE *o)
+{
+    char *s=(char *) malloc(sizeof(char)*maxLengthInstruction);
+    char *number=(char *) malloc(sizeof(char)*maxLengthInstruction);
+
+    while(fgets(s,maxLengthInstruction,f)!=NULL)
+    {
+        ///  printf("%s\n%s\n\n",s,strstr(s,"PRINT_L"));
+        //PRINT_ALLL();
+        //printf("%s\n",s);
+        int len=0;
+        if(s[0]=='A' && s[1]=='F')
+        {
+            int index=3;
+
+            while(isdigit(s[index]))
+            {
+                number[len++]=s[index++];
+            }
+            number[len++]=NULL;
+            /// printf("%d\n",atoi(number));
+            addElementAtFront(atoi(number));
+        }
+        else

```

Comment:
Gosh, this looks very ugly!
Please find a nicer way to parse the file, it's terribly simple - use `fscanf` or `strtok`